src/dso_api/dynamic_api/filters/backends.py

```diff
@@ -0,0 +1,84 @@
+"""Adapter classes for Django REST Framework.
+
+The "filter backend" classes follow an API of Django REST Framework,
+that allows adapting the QuerySet before the view/serializer starts.
+This gives a chance to add filtering and ordering to the queryset.
+"""
+from __future__ import annotations
+
+from rest_framework.filters import BaseFilterBackend, OrderingFilter
+
+from dso_api.dynamic_api.filters import openapi
+from dso_api.dynamic_api.filters.parser import QueryFilterEngine
+
+
+class DynamicFilterBackend(BaseFilterBackend):
+    """A backend that performs filtering backed on query-string parameters.
+
+    Usage in views::
+
+        class View(GenericAPIView):
+            filter_backends = [filters.DynamicFilterBackend]
+    """
+
+    def filter_queryset(self, request, queryset, view):
+        """Filter the queryset of the view.
+        This hook function is called by Django REST Framework as apply
+        this of this filter backend on the constructed queryset.
+        """
+        input_crs = getattr(request, "accept_crs", None)
+        engine = QueryFilterEngine(request.user_scopes, request.GET, input_crs)
+        return engine.filter_queryset(queryset)
+
+    def get_schema_operation_parameters(self, view):
+        """Generate the OpenAPI fragments for the filter parameters.
+        This hook function is called by Django REST Framework to generate
+        the OpenAPI data.
+        """
+        return openapi.get_table_filter_params(view.table_schema)
+
+
+class DynamicOrderingFilter(OrderingFilter):
+    """Ordering filter, following the DSO spec.
+
+    This adds an ``?_sort=<fieldname>,-<desc-fieldname>`` option to the view.
+    Usage in views::
+
+        class View(GenericAPIView):
+            filter_backends = [filters.DynamicOrderingFilter]
+    """
+
+    ordering_param = "_sort"  # Enforce DSO-specific name.
+
+    def get_ordering(self, request, queryset, view):
+        if self.ordering_param not in request.query_params:
+            # Allow DSO 1.0 Dutch "sorteer" parameter
+            # Can adjust 'self' as this instance is recreated each request.
+            if "sorteer" in request.query_params:
```

Comment:
I thought the ideas was to use `_` prefixes for these parameters, or has this been changed?

Reply:
This is for backwards compatibility with older versions of the DSO standard.